extensions/2.0/Khronos/KHR_meshopt_compression/README.md

```diff
@@ -0,0 +1,675 @@
+
+## Specifying compressed views
+
+As explained in the overview, this extension operates on bufferViews. This allows the loaders to directly decompress data into GPU memory and minimizes the JSON size impact of specifying compressed data. To specify the compressed representation, `KHR_meshopt_compression` extension section overrides the source buffer index as well as specifying the buffer parameters and a compression mode/filter (detailed later in the specification):
+
+```json
+{
+	"buffer": 1,
+	"byteOffset": 0,
+	"byteLength": 2368,
+	"byteStride": 16,
+	"target": 34962,
+	"extensions": {
+		"KHR_meshopt_compression": {
+			"buffer": 0,
+			"byteOffset": 1024,
+			"byteLength": 347,
+			"byteStride": 16,
+			"mode": "ATTRIBUTES",
+			"count": 148
+		}
+	}
+}
+```
+
+In this example, the uncompressed buffer contents is stored in buffer 1 (this can be used by loaders that don't implement this extension). The compressed data is stored in a separate buffer, specifying a separate byte range (with compressed data). Note that for compressors to work, they need to know the compression `mode`, `filter` (for `"ATTRIBUTES"` mode), and additionally the layout of the encoded data - `count` elements with `byteStride` bytes each. This data is specified in the extension JSON; while in some cases `byteStride` is available on the parent `bufferView` declaration, JSON schema prohibits specifying this for some types of storage such as index data.
+
+## JSON schema updates
+
+Each `bufferView` can contain an extension object with the following properties:
+
+| Property | Type | Description | Required |
+|:---------|:--------------|:------------------------------------------| :--------------------------|
+| `buffer` | `integer` | The index of the buffer with compressed data. | :white_check_mark: Yes |
+| `byteOffset` | `integer` | The offset into the buffer in bytes. | No, default: `0` |
+| `byteLength` | `integer` | The length of the compressed data in bytes. | :white_check_mark: Yes |
+| `byteStride` | `integer` | The stride, in bytes. | :white_check_mark: Yes |
+| `count` | `integer` | The number of elements. | :white_check_mark: Yes |
+| `mode` | `string` | The compression mode. | :white_check_mark: Yes |
+| `filter` | `string` | The compression filter. | No, default: `"NONE"` |
+
+`mode` represents the compression mode using an enumerated value that must be one of `"ATTRIBUTES"`, `"TRIANGLES"`, `"INDICES"`.
+
+`filter` represents the post-decompression filter using an enumerated value that must be one of `"NONE"`, `"OCTAHEDRAL"`, `"QUATERNION"`, `"EXPONENTIAL"`, `"COLOR"`.
+
+For the extension object to be valid, the following must hold:
+
+- When parent `bufferView` has `byteStride` defined, it matches `byteStride` in the extension JSON
+- The parent `bufferView.byteLength` is equal to `byteStride` times `count`
```

Comment:
The base spec allows omitting the final trailing bytes when the stride value is greater than the accessor element size. As currently written, such accessors would be incompatible with the extension.

Reply:
Ideally this should be a part of testing suite for tools. So if an encoder encounters such an asset and it preserves the uncompressed data, it must add the trailing bytes to the latter.